src/extension.ts

```diff
@@ -162,39 +144,56 @@ async function runWarPackage(tomcatController: TomcatController, uri?: vscode.Ur
     await runOnTomcat(tomcatController, false, uri);
 }
 
-async function selectServer(ui: VSCodeUI, placeHolder: string, tomcatController: TomcatController, withNew?: string): Promise<string | undefined> {
+async function selectServer(tomcatController: TomcatController, tomcatServer?: TomcatServer): Promise<TomcatServer> {
+    if (tomcatServer) {
+        return tomcatServer;
+    }
     const serverSet: TomcatServer[] = tomcatController.getServerSet();
-    let serverPick: PickWithData<string> | undefined;
-    let serverPicks: PickWithData<string>[] = [];
-
-    if (serverSet && serverSet.length !== 0) {
-        serverPicks = serverPicks.concat(serverSet.map((server: TomcatServer) =>
-            new PickWithData(server.getName(), server.getName())));
-        if (withNew) {
-            serverPicks.push(new PickWithData(withNew, DialogMessage.createServer));
+    const pick: vscode.QuickPickItem = await vscode.window.showQuickPick(
+        [...serverSet, { label: `$(plus) ${DialogMessage.createServer}`, description: null }],
```

Comment:
better to write something to the description. not set it to 'null'

Reply:
description will show up in the dialog, not what we are expecting. Or we could set it empty.